cmake/macros.cmake

```diff
@@ -156,8 +109,63 @@ if(APPLE)
   endif()
 endif()
 
-if(CMAKE_VERSION VERSION_LESS 3.13.0)
-  macro(target_link_directories _TARGET _SCOPE)
-    link_directories(${ARGN})
-  endmacro()
-endif()
+function(SDL_PrintSummary)
+  ##### Info output #####
+  message(STATUS "")
+  message(STATUS "SDL3 was configured with the following options:")
+  message(STATUS "")
+  message(STATUS "Platform: ${CMAKE_SYSTEM}")
+  message(STATUS "64-bit:   ${ARCH_64}")
+  message(STATUS "Compiler: ${CMAKE_C_COMPILER}")
+  message(STATUS "Revision: ${SDL_REVISION}")
+  message(STATUS "Vendor:   ${SDL_VENDOR_INFO}")
+  message(STATUS "")
+  message(STATUS "Subsystems:")
+
+  find_stringlength_longest_item(SDL_SUBSYSTEMS maxLength)
+  foreach(_SUB IN LISTS SDL_SUBSYSTEMS)
+    string(LENGTH ${_SUB} _SUBLEN)
+    math(EXPR _PADLEN "(${maxLength} + 1) - ${_SUBLEN}")
+    string(RANDOM LENGTH ${_PADLEN} ALPHABET " " _PADDING)
+    string(TOUPPER ${_SUB} _OPT)
+    message_bool_option(${_SUB} SDL_${_OPT} ${_PADDING})
+  endforeach()
+  message(STATUS "")
+  message(STATUS "Options:")
+  list(SORT ALLOPTIONS)
+  message_dictlist(ALLOPTIONS)
+  message(STATUS "")
+  message(STATUS " Build Shared Library: ${SDL_SHARED}")
+  message(STATUS " Build Static Library: ${SDL_STATIC}")
+  if(SDL_STATIC)
+    message(STATUS " Build Static Library with Position Independent Code: ${SDL_STATIC_PIC}")
+  endif()
+  if(APPLE)
+    message(STATUS " Build libraries as Apple Framework: ${SDL_FRAMEWORK}")
+  endif()
+  message(STATUS "")
+  if(UNIX)
+    message(STATUS "If something was not detected, although the libraries")
+    message(STATUS "were installed, then make sure you have set the")
+    message(STATUS "CMAKE_C_FLAGS and CMAKE_PREFIX_PATH CMake variables correctly.")
+    message(STATUS "")
+  endif()
+
+  if(WARN_ABOUT_ARM_SIMD_ASM_MIT)
+    message(STATUS "SDL is being built with ARM SIMD optimizations, which")
+    message(STATUS "uses code licensed under the MIT license. If this is a")
+    message(STATUS "problem, please disable that code by rerunning CMake with:")
+    message(STATUS "")
+    message(STATUS "    -DSDL_ARMSIMD=OFF")
+    message(STATUS "")
+  endif()
+
+  if(WARN_ABOUT_ARM_NEON_ASM_MIT)
+    message(STATUS "SDL is being built with ARM NEON optimizations, which")
+    message(STATUS "uses code licensed under the MIT license. If this is a")
+    message(STATUS "problem, please disable that code by rerunning CMake with:")
+    message(STATUS "")
+    message(STATUS "    -DSDL_ARMNEON=OFF")
+    message(STATUS "")
+  endif()
+endfunction()
```

Comment:
Missing newline at end of file.

P.S.: Is it intentional that you keep individual commit messages from squashing?

Reply:
I squashed all commits today because it became a pain to repeatedly rebase all commits.

Reply:
Squashing is good, but reading the commit message makes one's eyes burn

Reply:
Oh, I'll rewrite it.